### PR TITLE
Updating graphql-spring-boot to use graphql-java:15.  

### PR DESCRIPTION
**Pull request by setchy**
Updating to use graphql-java:15

Releasing graphql-spring-boot will be dependent on the release of the following dependencies

**graphql-java-servlet** 
- PR - https://github.com/graphql-java-kickstart/graphql-java-servlet/pull/270
- Enhancement Issue - https://github.com/graphql-java-kickstart/graphql-java-servlet/issues/269
- For now I have made the assumption that when released the new version would be 9.3.0 (currently set to 9.3.0-SNAPSHOT).  Will update `gradle.properties` with the final version once released

**graphql-java-annotations** 
- PR - https://github.com/Enigmatis/graphql-java-annotations/pull/260
- Enhancement Issue - https://github.com/Enigmatis/graphql-java-annotations/issues/259
- For now I have made the assumption that when released the new version would be 8.2 (currently set to 8.2-SNAPSHOT).  Will update `gradle.properties` with the final version once released

### CODE REVIEW

**Comment by BlasiusSecundus** [comment]
I'm afraid, this is not so easy:) GraphQL Java 15 (https://github.com/graphql-java/graphql-java/releases/tag/v15.0) has some breaking changes as well as new features, that should be properly integrated.

The heavy lifting, I guess, is largely done in https://github.com/graphql-java-kickstart/graphql-java-tools. I see there is an aptly named branch there (https://github.com/graphql-java-kickstart/graphql-java-tools/tree/405-upgrade-to-graphql-java-15), but without commits. Perhaps @vojtapol can tell more about the status/progress of this feature.

As for the recently added GraphQL Annotations support, it is also on GraphQL Java 14 and can only be upgraded once the GraphQL Java Annotations library itself is updated.

**Comment by setchy** [comment]
@BlasiusSecundus - maybe I'm missing something, but locally with the updates in https://github.com/graphql-java-kickstart/graphql-java-servlet/pull/270 and in this PR, everything compiled, all tests passed and my GraphQL SpringBoot services (~10) functioned as expected.

**Comment by BlasiusSecundus** [comment]
Yes, you are totally right, java-graphql-tools was already updated, I missed that.

**Comment by setchy** [comment]
If this https://github.com/graphql-java-kickstart/graphql-java-servlet/pull/270 could be merged / released, then I could update this branch for graphql-spring-boot and see the true state of the CI test results

**Comment by setchy** [comment]
Issue opened for graphql-java-annotations https://github.com/Enigmatis/graphql-java-annotations/issues/259

**Comment by setchy** [comment]
Pull Request https://github.com/Enigmatis/graphql-java-annotations/pull/260

File: gradle.properties [hunk 1]
```diff
@@ -39,10 +39,10 @@ TARGET_COMPATIBILITY = 1.8
 
 ### Dependencies
 
-LIB_GRAPHQL_JAVA_VER = 14.1
-LIB_SPRING_BOOT_VER = 2.3.1.RELEASE
-LIB_GRAPHQL_SERVLET_VER = 9.2.0
-LIB_GRAPHQL_JAVA_TOOLS_VER = 6.1.0
+LIB_GRAPHQL_JAVA_VER = 15.0
+LIB_SPRING_BOOT_VER = 2.3.4.RELEASE
+LIB_GRAPHQL_SERVLET_VER = 9.3.0-SNAPSHOT
```
Comment (m1ngyuan) [inline gradle.properties:44]: Maybe it is 9.2.0-SNAPSHOT ?

Comment (setchy) [reply gradle.properties:44]: graphql-java-servlet 9.2.0 is already released - https://github.com/graphql-java-kickstart/graphql-java-servlet/releases

The next working version would be 9.3.0-SNAPSHOT

**Comment by setchy** [comment]
graphql-java-annotations v8.2 has now been released, with support for graphql-java-tools-15.0

https://github.com/Enigmatis/graphql-java-annotations/releases/tag/v8.2

File: gradle.properties (hunk 1 above)
Comment (setchy) [reply gradle.properties:44]: Has now been merged and released at graphql-java-servlet 10.0.0

https://github.com/graphql-java-kickstart/graphql-java-servlet/releases/tag/v10.0.0

**Comment by setchy** [comment]
graphql-java-servlets v10.0.0 has now been released, with support for graphql-java-tools-15.0

https://github.com/graphql-java-kickstart/graphql-java-servlet/releases/tag/v10.0.0

**Comment by setchy** [comment]
@oliemansm @vojtapol - all the dependent libraries are now graphql-java-15 compatible:

- ([graphql-java-annotations](https://github.com/Enigmatis/graphql-java-annotations/releases/tag/v8.2)
- [graphql-java-servlet](https://github.com/graphql-java-kickstart/graphql-java-servlet/releases/tag/v10.0.0))

Are you able to do a review of this PR and merge/release at your convenience :)

**Comment by oliemansm** [comment]
Thanks @setchy ! Once all checks have passed I'll merge it.

**Comment by oliemansm** [comment]
@setchy Can't merge it as is unfortunately since there are failing unit tests. First glance I notice something about Scalar Long not found, which rings a bell being mentioned in the release notes of graphql-java 15.0: https://github.com/graphql-java/graphql-java/releases/tag/v15.0.

**Comment by setchy** [comment]
> @setchy Can't merge it as is unfortunately since there are failing unit tests. First glance I notice something about Scalar Long not found, which rings a bell being mentioned in the release notes of graphql-java 15.0: https://github.com/graphql-java/graphql-java/releases/tag/v15.0.

thanks @oliemansm - let me look into this.  

I believe I could address this in a few ways
- add [graphql-java-extended-scalars v1.1.0](https://github.com/graphql-java/graphql-java-extended-scalars/releases/tag/1.1.0) as a dependency to keep the tests as-is
- remove uses of any 'non standard scalars' as defined on [here](https://github.com/graphql-java/graphql-java/releases/tag/v15.0)

Do you have a preference?  Personally I'm thinking of the first path

**Comment by oliemansm** [comment]
@setchy By the looks of it that version hasn't been released to maven central yet, see https://github.com/graphql-java/graphql-java-extended-scalars/issues/24. It also pulls in a lot of other scalars of which some might be conflicting (e.g. DateTime) with existing usages. While the DateTime scalar implemented by that library seems lacking functionality compared to https://github.com/donbeave/graphql-java-datetime (see also https://github.com/graphql-java/graphql-java-extended-scalars/issues/7).

So considering all that and mainly the maven central sync issue I'd go with your option 2 in the end.

**Comment by setchy** [comment]
Awesome - by far the easiest path.

Should be all sorted now @oliemansm

**Comment by oliemansm** [comment]
Thanks @setchy!

**Pull request merged**